libraries/session-storage/impl/src/main/kotlin/io/element/android/libraries/sessionstorage/impl/DatabaseSessionStore.kt

```diff
@@ -67,30 +71,38 @@ class DatabaseSessionStore @Inject constructor(
             return
         }
 
-        // Copy new data from SDK, but keep login timestamp
-        database.sessionDataQueries.updateSession(
-            sessionData.copy(
-                loginTimestamp = result.loginTimestamp,
-            ).toDbModel()
-        )
+        sessionDataMutex.withLock {
```

Comment:
I would wrap the entire function body with the Mutex, to include the db query. But maybe there is a good reason not to do so?